### PR TITLE
Move towards standardized project layout/setup

### PR DESCRIPTION
**Pull request by wlach**
* Remove setup.py, replace with pyproject.toml
* Use flit for building package/wheels
* Move project logic to `src/`
* Move tests to `tests/`

Layout/ideas borrowed from flask-debugtoolbar

### CODE REVIEW

File: tests/test_basic.py [hunk 1]
```diff
@@ -8,11 +8,12 @@
 from unittest import mock
 
 from flask import Flask
+from speaklater import make_lazy_string
+
```
Comment (wlach) [inline tests/test_basic.py:12]: I think moving things to `src/` made ruff realize that `flask_mail` was a 1st party package

File: pyproject.toml [hunk 2]
```diff
@@ -1,5 +1,40 @@
+[project]
+name = "Flask-Mail"
+version = "0.9.1"
+description = "Flask extension for sending email"
+readme = "README.md"
+authors = [{ name = "Dan Jacob", email = "danjac354@gmail.com" }]
+maintainers = [{ name = "Pallets Ecosystem", email = "contact@palletsprojects.com" }]
```
Comment (wlach) [inline pyproject.toml:7]: Seemed to make sense to change this (modelled after flask-debugtoolbar, giving credit to the original author while making it clear who's the active maintainer)

File: pyproject.toml [hunk 3]
```diff
@@ -1,5 +1,40 @@
+[project]
+name = "Flask-Mail"
+version = "0.9.1"
+description = "Flask extension for sending email"
+readme = "README.md"
+authors = [{ name = "Dan Jacob", email = "danjac354@gmail.com" }]
```
Comment (davidism) [inline pyproject.toml:6]:
```suggestion
authors = [{ name = "Dan Jacob" }]
```

File: pyproject.toml [hunk 4]
```diff
@@ -1,5 +1,40 @@
...
+    "Development Status :: 4 - Beta",
+    "Environment :: Web Environment",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+    "Topic :: Internet :: WWW/HTTP :: Dynamic Content",
+    "Topic :: Software Development :: Libraries :: Python Modules",
+]
+dependencies = ["Flask", "blinker"]
```
Comment (davidism) [inline pyproject.toml:18]:
```suggestion
dependencies = [
    "flask",
    "blinker",
]
```

File: pyproject.toml [hunk 5]
```diff
@@ -1,5 +1,40 @@
...
+license = { text = "BSD" }
```
Comment (davidism) [inline pyproject.toml:19]:
```suggestion
license = { file = "LICENSE" }
```

File: pyproject.toml [hunk 6]
```diff
@@ -1,5 +1,40 @@
...
+requires-python = ">=3.8"
+
+[project.urls]
+Homepage = "https://github.com/pallets-eco/flask-mail"
```
Comment (davidism) [inline pyproject.toml:23]:
```suggestion
Documentation = "https://flask-mail.readthedocs.io"
Changes = "https://flask-mail.readthedocs.io/en/latest/changes/"
Source = "https://github.com/pallets-eco/flask-mail/"
Chat = "https://discord.gg/pallets"
```

**Pull request merged**